tests/basic-test.sh

```diff
@@ -459,6 +459,32 @@ assert_file_has_content checkout-test-union-add/union-add-test 'existing file fo
 assert_file_has_content checkout-test-union-add/union-add-test2 'another file for union add testing'
 echo "ok checkout union add"
 
+# Create some new files for testing
+cd ${test_tmpdir}
+mkdir add-dirs-test
+mkdir add-dirs-test/test_one
+chmod a+w add-dirs-test/test_one
+echo 'file for add dirs testing' > add-dirs-test/test_one/test_file
+$OSTREE commit ${COMMIT_ARGS} -b test-add-dirs --tree=dir=add-dirs-test
+$OSTREE checkout --add-dirs test-add-dirs checkout-test-add-dirs
+echo "ok adding new directories and new file"
+# Make a new file, and try the checkout again
+echo 'second test file' >  add-dirs-test/test_one/test_second_file
+$OSTREE commit ${COMMIT_ARGS} -b test-add-dirs --tree=dir=add-dirs-test
+# Check out the latest commit, should fail due to existing files
+if $OSTREE checkout --add-dirs test-add-dirs checkout-test-add-dirs 2> err.txt; then
+    assert_not_reached "checking out files unexpectedly succedded!"
+fi
+assert_file_has_content err.txt 'File exists'
+# Verify that Union mode still functions properly
+# First 'break' the hard link for the previous files
+rm checkout-test-add-dirs/test_one/test_file
```

Comment:
without this removing line, the `test_file` content will not be affected when union is performed